shared/js/background/events/referrer-trimming.js

```diff
@@ -3,18 +3,24 @@ const trackerutils = require('../tracker-utils')
 const utils = require('../utils.es6')
 const browserName = utils.getBrowserName()
 
+/**
+ * @typedef Referrer
+ * @property {string} site
+ *   The referrer URL.
+ * @property {string} referrerHost
+ *   The referrer host.
+ * @property {string} referrer
+ *   The truncated referrer.
+ */
+
 /**
  * @param {{tabId: number, url: string, requestHeaders: Array<{name: string, value:string}>}} e
  *
  * @returns {{requestHeaders: Array<{name: string, value:string}>} | { redirectUrl: URL } | undefined}
  */
 module.exports = function limitReferrerData (e) {
-    let referrer = e.requestHeaders.find(header => header.name.toLowerCase() === 'referer')
-    if (referrer) {
-        referrer = referrer.value
-    } else {
-        return
-    }
+    const referrer = e.requestHeaders.find(header => header.name.toLowerCase() === 'referer')?.value
```

Comment:
Was this needed, what it was previously is much more readable to me. (I'd rather not get into opinionated reviews just interested why it was changed)

Reply:
Actually ignore this the other side is equally suboptimal because of shadowing and I'm guessing TS wasn't happy there.

In general I prefer more temporary vars over chaining like this.

Will close though.